### PR TITLE
docs: add RFC Stages to our contributing guidelines

### PR DESCRIPTION
**Pull request by smoya**
This pull request improves the contributing guideline by introducing a structured framework for proposing and implementing changes to our source code.

The RFC process ensures that significant changes are proposed, discussed, and implemented systematically, reducing miscommunication and aligning the community efforts.

Strongly inspired by [AsyncAPI contributing guidelines](https://github.com/asyncapi/spec/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by adolsalamanca — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,100 +1,90 @@
...
 
-* Create and manage pull request:
+### Creating a Pull Request:
 
-    * [Create a pull request using GitHub](https://help.github.com/articles/creating-a-pull-request).
-      If you know a core developer well suited to reviewing your pull
-      request, either mention them (preferably by GitHub name) in the PR's
-      body or [assign them as a reviewer](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/).
+Once your RFC is accepted (and in the case you didn't open a PR yet), or for smaller changes that do not require the RFC process, you are ready to open a PR.
+Please estrictly follow the guidelines above:
```
Comment (JamesGuthrie) [inline CONTRIBUTING.md:74]:
```suggestion
Please strictly follow the guidelines below:
```

Comment (smoya) [reply CONTRIBUTING.md:74]: LOL! And here is the proof that there are still people who dont write docs exclusively using ChatGPT (note the "exclusively" here 😆 )

Comment (smoya) [reply CONTRIBUTING.md]: Fixed BTW. thks!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,100 +1,90 @@
...
-sign the [Contributor License Agreement](https://cla-assistant.io/timescale/pgai) (CLA). By signing a CLA, we can
-ensure that the community is free and confident in its ability to use your
-contributions.
+We appreciate any help the community can provide to make pgai better!
 
-## Development
+## How to Contribute
 
-Please follow our DEVELOPMENT doc for [instructions how to develop and test](https://github.com/timescale/pgai/blob/main/DEVELOPMENT.md).
+### Reporting Issues
```
Comment (billy-the-fish) [inline CONTRIBUTING.md:9]:
```suggestion
### Report Issues
```
We try to avoid -ing words in titles.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,100 +1,90 @@
...
+If you encounter any bugs or have suggestions for improvements, please submit an issue through the [GitHub issue tracker](https://github.com/timescale/pgai/issues). Include as much detail as possible to help us understand and resolve the problem.
 
-## Code review workflow
+### Proposing Features
```
Comment (billy-the-fish) [inline CONTRIBUTING.md:12]:
```suggestion
### Propose Features
```
We try to avoid -ing words in titles.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,100 +1,90 @@
...
 
-    * Address feedback by amending your commit(s). If your change contains
-      multiple commits, address each piece of feedback by amending that
-      commit to which the particular feedback is aimed.
+1. Fork the repository and create a new branch for your work.
+2. Write clean, well-documented, and tested code.
+3. Ensure that your code adheres to the project's style and guidelines. 
+4. Include relevant tests and documentation updates.
+5. Don't mix bug fixes with unrelated feature enhancements or stylistical changes.
+6. The project uses [conventional commits][conventional-commits]. It's enforced by CI, you won't be able to merge PRs if your commits do not comply. This helps us automate the release process, changelog generation, etc. For example:
```
Comment (billy-the-fish) [inline CONTRIBUTING.md:81]: Please add the following to the bottom of the page:

[conventional-commits]: https://www.conventionalcommits.org/en/v1.0.0/

**Review by billy-the-fish — APPROVED** [review]
Couple of small doc things. Nice page.

File: CONTRIBUTING.md (hunk 2 above)
Comment (smoya) [reply CONTRIBUTING.md]: Fixed

File: CONTRIBUTING.md (hunk 4 above)
Comment (smoya) [reply CONTRIBUTING.md:81]: You mean at the bottom of the Create a Pull Request section?

Comment (smoya) [reply CONTRIBUTING.md]: link fixed 👍

**Pull request merged**